File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/templatemethod/transport/Plane.java
// Plane.java

package com.bigboxcode.designpattern.templatemethod.transport;

public class Plane  extends Transport {
    @Override
    void createBody() {
        System.out.println("Creating Plane Body");
    }

    @Override
    void addEngine() {
        System.out.println("Adding Engine to Plane");
    }

    @Override
    void addWheel() {
        System.out.println("Adding 3 Wheels to Plane");
    }

    @Override
    void addWing() {
        System.out.println("Adding Wings Plane");
    }
}

File: design-patterns/php/templatemethod/transport/Plane.php
<?php
// Plane.php

namespace BigBoxCode\DesignPattern\TemplateMethod\Transport;

class Plane extends Transport {
    public function createBody(): void {
        echo "Creating Plane Body\n";
    }

    public function addEngine(): void {
        echo "Adding Engine to Plane\n";
    }

    public function addWheel(): void {
        echo "Adding 3 Wheels to Plane\n";
    }

    public function addWing(): void {
        echo "Adding Wings Plane\n";
    }
}